pkg/kubelet/nodestatus/setters.go

```diff
@@ -402,8 +404,13 @@ func VersionInfo(versionInfoFunc func() (*cadvisorapiv1.VersionInfo, error), //
 		node.Status.NodeInfo.ContainerRuntimeVersion = fmt.Sprintf("%s://%s", runtimeTypeFunc(), runtimeVersion)
 
 		node.Status.NodeInfo.KubeletVersion = version.Get().String()
-		// TODO: kube-proxy might be different version from kubelet in the future
-		node.Status.NodeInfo.KubeProxyVersion = version.Get().String()
+
+		if utilfeature.DefaultFeatureGate.Enabled(features.DisableNodeKubeProxyVersion) {
+			node.Status.NodeInfo.KubeProxyVersion = ""
```

Comment:
Let's add a comment so someone in the future doesn't come along and optimize that out...

```suggestion
			// This field is deprecated and should be cleared if it was previously set.
			node.Status.NodeInfo.KubeProxyVersion = ""
```